config/config.go

```diff
@@ -617,6 +624,7 @@ type RemoteWriteConfig struct {
 	// values arbitrarily into the overflow maps of further-down types.
 	HTTPClientConfig config_util.HTTPClientConfig `yaml:",inline"`
 	QueueConfig      QueueConfig                  `yaml:"queue_config,omitempty"`
+	MetadataConfig   MetadataConfig               `yaml:"metadata_config,omitempty"`
```

Comment:
If you make this a pointer, and it is empty: It can replace the boolean, right? (:

Reply:
Sorry, I could have been clearer. The problem is not about the code change itself but the philosophy behind the change. 

We'd like integrations to opt-out as opposed to opt-in. Empty configurations would imply this is not enabled, or am I missing something?

Reply:
I am ok with the boolean, this is experimental and we have removed remote config fields before without issue. 

The other option I am thinking is that setting the send interval to 0 could turn this off, but I think I would prefer the explicit flag.

Reply:
> We'd like integrations to opt-out as opposed to opt-in. Empty configurations would imply this is not enabled, or am I missing something?

Again. If you make this a pointer you don't need  boolean. If user does not set it up it will be nil, so you will know it's opt-out.

Reply:
Barket, I'm awfully sorry. I'm still not following your comment if we want users to get this enabled by default. I don't see how a pointer makes a difference here.

Reply:
Sorry i missed that you want opt-out not opt-in...

Why you want to introduce opt-out? This is a breaking change then, no? Also how this is experimental if it is enabled by default. Not all servers would even implement this.

Reply:
Sorry I might miss that decision but to me default turn on is not great. The metadata has real implications especially as it's sends together now, so enabling it even though proto is forward compatible sounds weird to me... Happy to discuss it though.

Reply:
As you mentioned, given the guarantee of protobuf, this wouldn't be a breaking change. Consumers that do not implement the message, would simply discard it.

Given that, I don't see any harm done on turning this on by default and enabling our users to get the functionality for free on an update. That being said, if you have a _very_ strong opinion about disabling it -- I'm happy to comply and use that pointer.

Reply:
My view on this is that eventually I would like remote write to send all samples and metadata together by default, but provide users a mechanism to turn off parts they do not use.

However, we could be conservative and default to off for a release or two while a few early adopters turn it on to see what the bandwidth increase looks like in a variety of workloads. If bandwidth is reasonable we can change the default then. WDYT?

Reply:
For this particular question given the bandwidth numbers proposed so far, I'd say keep it simple and hardcode to on. If you wanted to be efficient you wouldn't be using remote write ;)

Reply:
Haha fair. You convinced me. LGTM (:

Reply:
Thank you all! Resolving this on then.